### DISCUSSION TITLE
It's just me or trpc + zod-prisma-types makes typescript types loading too slow

### DISCUSSION THREAD

**Discussion by CursedMun** [trpc/trpc#4758] [Q&A]
At the start of my monorepo project everything were fine, but after adding around 20-40 endpoints it everytime I'm trying to see some type when calling trpc the types take about 10-30 seconds to load.

Is it a bug or it's just me?

**Comment by max-lt** [comment]
It' not you, probably the same problem than https://github.com/trpc/trpc/discussions/2448

**Comment by dBianchii** [comment]
Hi, I'm having problems using tRPC because of typescript performance reasons. Any suggestions? I am reading the article https://github.com/trpc/trpc/discussions/2448 but had no luck

**Reply by dBianchii** [reply]
![image](https://github.com/trpc/trpc/assets/17372417/4ea0b58b-715b-436b-8eae-a90627500bfc)
I want to cry :crying_cat_face:

**Reply by CursedMun** [reply]
I did manage to overcome this by extracting trpc to some sort of a library where I transpile everything to static TS which helped performance wise alot.

**Reply by dBianchii** [reply]
Well, I have this exact same setup, where I call build in a package, and it outputs the types to ./dist. But still, I cant even run that command anymore because I have this error from the picture I sent

**Reply by CursedMun** [reply]
you probably have too many redundancy in you're db models

**Reply by CursedMun** [reply]
try to make them easier to ready, remove some pointing to the models, maybe it will help a little bit

**Reply by dBianchii** [reply]
@CursedMun I don't think I'm doing anything strange for the db. In case you want to check, I have the files here: https://github.com/dBianchii/kodix-turbo/tree/main/packages/db/src/schema
